internal/security/secretstore/tokenprovider_test.go

```diff
@@ -18,78 +18,123 @@ package secretstore
+	"github.com/stretchr/testify/mock"
 )
 
 func TestInvalidProvider(t *testing.T) {
 	config := secretstoreclient.SecretServiceInfo{
 		TokenProvider:     "does-not-exist",
 		TokenProviderType: OneShotProvider,
 	}
-	cancel, err := testCommon(config)
+	mockExecRunner := mockExecRunner{}
+	mockExecRunner.On("LookPath", "does-not-exist").Return("", errors.New("fake file does not exist"))
+	cancel, err := testCommon(config, &mockExecRunner)
 	defer cancel()
 	assert.NotNil(t, err)
+	mockExecRunner.AssertExpectations(t)
 }
 
 func TestInvalidProviderType(t *testing.T) {
 	config := secretstoreclient.SecretServiceInfo{
-		TokenProvider:     "/bin/true",
+		TokenProvider:     "success-executable",
 		TokenProviderType: "simple",
 	}
-	cancel, err := testCommon(config)
+	mockExecRunner := mockExecRunner{}
+	cancel, err := testCommon(config, &mockExecRunner)
 	defer cancel()
 	assert.Error(t, err)
+	mockExecRunner.AssertExpectations(t)
 }
 
 func TestNoConfig(t *testing.T) {
 	ctx, cancel := context.WithCancel(context.Background())
-	p := NewTokenProvider(ctx, logger.MockLogger{})
+	p := NewTokenProvider(ctx, logger.MockLogger{}, ExecWrapper{})
+	// don't call SetConfiguration()
 	err := p.Launch()
 	defer cancel()
 	assert.Error(t, err)
 }
 
-func TestTrue(t *testing.T) {
+func TestSuccess(t *testing.T) {
 	config := secretstoreclient.SecretServiceInfo{
-		TokenProvider:     "/bin/true",
+		TokenProvider:     "success-executable",
 		TokenProviderType: OneShotProvider,
+		TokenProviderArgs: []string{"arg1", "arg2"},
 	}
-	cancel, err := testCommon(config)
+	mockExecRunner := mockExecRunner{}
+	mockCmd := mockCmd{}
+	mockExecRunner.On("LookPath", "success-executable").Return("success-executable", nil)
+	mockExecRunner.On("CommandContext", mock.Anything, "success-executable", []string{"arg1", "arg2"}).Return(&mockCmd)
+	mockCmd.On("Start").Return(nil)
+	mockCmd.On("Wait").Return(nil)
+	cancel, err := testCommon(config, &mockExecRunner)
 	defer cancel()
 	assert.Nil(t, err)
+	mockExecRunner.AssertExpectations(t)
+	mockCmd.AssertExpectations(t)
 }
 
-func TestFalse(t *testing.T) {
+func TestFailure(t *testing.T) {
 	config := secretstoreclient.SecretServiceInfo{
-		TokenProvider:     "/bin/false",
+		TokenProvider:     "failure-executable",
 		TokenProviderType: OneShotProvider,
+		TokenProviderArgs: []string{"arg1", "arg2"},
 	}
-	cancel, err := testCommon(config)
+	mockExecRunner := mockExecRunner{}
+	mockCmd := mockCmd{}
+	mockExecRunner.On("LookPath", "failure-executable").Return("failure-executable", nil)
+	mockExecRunner.On("CommandContext", mock.Anything, "failure-executable", []string{"arg1", "arg2"}).Return(&mockCmd)
+	mockCmd.On("Start").Return(nil)
+	mockCmd.On("Wait").Return(&exec.ExitError{ProcessState: &os.ProcessState{}})
+	cancel, err := testCommon(config, &mockExecRunner)
 	defer cancel()
 	assert.Error(t, err)
+	mockExecRunner.AssertExpectations(t)
+	mockCmd.AssertExpectations(t)
 }
 
-func TestEcho(t *testing.T) {
```

Comment:
Would this be valuable in the new Linux file?

Reply:
The original intent of TestEcho was to check command line argument processing.  `TestCreatesFile()` does the equivalent  testing.